src/main.ts

```diff
@@ -0,0 +1,51 @@
+import '@endo/init';
+import { iterateLatest, makeFollower, makeLeader } from '@agoric/casting';
+
+const urlSearchParams = new Proxy(new URLSearchParams(window.location.search), {
+  get: (searchParams, prop) => searchParams.get(prop.toString()),
+}) as { network?: string };
+
+const networkConfigUrl = (() => {
+  const agoricNetName = urlSearchParams.network ?? 'main';
+
+  if (agoricNetName === 'local') {
+    return 'https://wallet.agoric.app/wallet/network-config';
+  } else {
+    return `https://${agoricNetName}.agoric.net/network-config`;
+  }
+})();
+
+const leader = makeLeader(networkConfigUrl);
+const castingSpec = ':published.vaultFactory.governance';
+const follower = makeFollower(castingSpec, leader);
+
+type Params = { current: { EndorsedUI: { type: 'string'; value: string } } };
+
+const tryRedirect = async () => {
+  let endorsedUI;
+  for await (const { value } of iterateLatest<{ value: Params }>(follower)) {
```

Comment:
this loop for the head is awkward. I don't know a better way to do it with Casting though.

Reply:
Yea. I tried not using casting but couldn't reverse engineer the queries like I could with vstorage keys, seems a little more complicated. It really increases the bundle size by a lot unfortunately also.

Reply:
Like what size to what size? I wonder if it's something to come back to after release.

Reply:
Currently:
```
dist/index.html                     0.80 kB
dist/assets/index-e9cc255b.css      5.07 kB │ gzip:   1.78 kB
dist/assets/index-1759c800.js   2,193.05 kB │ gzip: 537.35 kB
```

Without casting:
```
dist/index.html                  0.80 kB
dist/assets/index-e9cc255b.css   5.07 kB │ gzip:  1.78 kB
dist/assets/index-6bb300c6.js   67.54 kB │ gzip: 22.94 kB
```

Without casting or endo:
```
dist/index.html                 0.80 kB
dist/assets/index-e9cc255b.css  5.07 kB │ gzip: 1.78 kB
dist/assets/index-e8fa93d7.js   1.05 kB │ gzip: 0.59 kB (probably closer to 2kB since I'd need to add some functions for the queries)
```

Reply:
I'm going to try again without casting, I think https://github.com/p2p-org/p2p-agoric-vstorage-viewer should be helpful